diff --git a/dcmgr/lib/dcmgr/models/hooks/vnet/vnet_hook.rb b/dcmgr/lib/dcmgr/models/hooks/vnet/vnet_hook.rb
index 75b8d0e..64d859a 100644
--- a/dcmgr/lib/dcmgr/models/hooks/vnet/vnet_hook.rb
+++ b/dcmgr/lib/dcmgr/models/hooks/vnet/vnet_hook.rb
@@ -25,8 +25,7 @@ Dcmgr::Models::Network.after_create do |network|
     :ipv4_network => :ipv4_network,
     :ipv4_prefix => :prefix,
     :domain_name => :domain_name,
-    :network_mode => 'virtual',
-    :editable => :editable
+    :network_mode => 'virtual'
   })
 end
 
